### PR TITLE
monolog-bundle: reconfigure deprecation handler

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| License       | MIT
| Doc issue/PR  | symfony/symfony-docs#...

- merges prod/deprecation.yaml into prod/monolog.yaml (it eases maintenance IMHO)
- moves deprecation channel to base env (see https://github.com/symfony/symfony/issues/35575#issuecomment-1020930037, it allows to handle deprecation channel consistently across all envs)

eg. exclude deprecation channel for console handler in dev, or persist deprecations seperate in test env.

When enabling `Debug::enable()` in `tests/bootstrap.php` we can even display deprecation on screen during phpunit, without relying on the bridge :)

WDYT?

### CODE REVIEW

**Comment by fabpot** [comment]
I would merge all configs in one file, see #1044

**Comment by nicolas-grekas** [comment]
Please add a conflict rule like in https://github.com/doctrine/DoctrineBundle/issues/1474

File: symfony/monolog-bundle/3.7/config/packages/monolog.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# As of Symfony 5.1, deprecations are logged in the dedicated "deprecation" channel when it exists
+#monolog:
+#    channels:
+#        - deprecation
```
Comment (nicolas-grekas) [inline symfony/monolog-bundle/3.7/config/packages/monolog.yaml:4]: let's un comment this?

File: symfony/monolog-bundle/3.7/config/packages/monolog.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                level: debug
+                formatter: monolog.formatter.json
+            console:
+                type: console
+                process_psr_3_messages: false
+                channels: ["!event", "!doctrine"]
+#            deprecation:
+#                type: stream
+#                channels: [deprecation]
+#                path: php://stderr
```
Comment (nicolas-grekas) [inline symfony/monolog-bundle/3.7/config/packages/monolog.yaml:62]: and uncomment this?

File: symfony/monolog-bundle/3.7/config/packages/monolog.yaml (hunk 1 above)
Comment (ro0NL) [reply symfony/monolog-bundle/3.7/config/packages/monolog.yaml]: :facepalm: :+1:

File: symfony/monolog-bundle/3.7/config/packages/monolog.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# As of Symfony 5.1, deprecations are logged in the dedicated "deprecation" channel when it exists
```
Comment (nicolas-grekas) [inline symfony/monolog-bundle/3.7/config/packages/monolog.yaml:1]: I would remove the "as of symfony 5.1" part

**Review by nicolas-grekas — APPROVED** [review]

File: symfony/monolog-bundle/3.7/config/packages/monolog.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                level: debug
+                channels: ["!event"]
+            # uncomment to get logging in your browser
+            # you may have to allow bigger header sizes in your Web server configuration
+            #firephp:
+            #    type: firephp
+            #    level: info
+            #chromephp:
+            #    type: chromephp
+            #    level: info
```
Comment (ro0NL) [inline symfony/monolog-bundle/3.7/config/packages/monolog.yaml:20]: is this common still, i use WDT 10/10 ...

File: symfony/monolog-bundle/3.7/config/packages/monolog.yaml (hunk 3 above)
Comment (ro0NL) [reply symfony/monolog-bundle/3.7/config/packages/monolog.yaml]: reformatted a bit, this should avoid merge conflicts on our side :angel:

**Review by fabpot — APPROVED** [review]

**Pull request merged**